### PR TITLE
CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by jeffmo**
Explains the structure of the repository and the steps needed to write one before sending a pull request.

(cc @nmn -- and of course anyone else -- for feedback)

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+You can write as many test files as you like as long as their names start with
+`test_`. 
+
+All test files that sit in the package version directory will be run by
+the test-runner for *all* versions of Flow the package version supports. If you 
+ever need to write a test for a particular version of Flow, you can put the 
+`test_` file in the appropriate flow version directory.
+
+#### 5) Run your tests with `./run_def_tests.sh left-pad`
```
Comment (bsr203) [inline CONTRIBUTING.md]: is this correct as I cannot locate any  run_def_tests.sh

If `runtests.sh` was the correct one, running like `runtests.sh left-pad`, ended up running all the packages, not just `left-pad`. Please verify.

the way it worked for moe was `cd cli && node dist/cli.js run-tests left-pad`

Comment (jeffmo) [inline CONTRIBUTING.md]: run_def_tests.sh is included in this diff :)

(Should be a little faster than the current runtests.sh)

Comment (bsr203) [inline CONTRIBUTING.md]: oops.. didn't notice it :-) thanks

Comment (nmn) [inline CONTRIBUTING.md]: Looks good!
A ASCII art file tree will probably be helpful.

**Comment by jeffmo** [comment]
An ASCII file structure outline is a good idea! I'll see if I can come up with something on Monday

**Pull request merged**